codegen-sbt/src/main/scala/caliban/codegen/CompileTimeCalibanPlugin.scala

```diff
@@ -0,0 +1,450 @@
+package caliban.codegen
+
+import caliban.codegen.CalibanSourceGenerator.TrackedSettings
+import caliban.tools.compiletime.Utils
+import sbt.Keys._
+import sbt.io.IO.defaultCharset
+import sbt.{ Compile, Def, Project, _ }
+
+import java.io.File
+import java.nio.channels.FileChannel
+import java.nio.file.{ Path, StandardOpenOption }
+import scala.util.Using
+
+object CompileTimeCalibanServerPlugin extends AutoPlugin {
+  override def requires = plugins.JvmPlugin
+  override def trigger  = noTrigger
+
+  object autoImport extends caliban.tools.compiletime.Config {
+
+    /* ## Tasks and settings namespace
+     *
+     * (https://www.scala-sbt.org/1.x/docs/Plugins-Best-Practices.html#Using+a+%E2%80%9Cmain%E2%80%9D+task+scope+for+settings)
+     */
+    lazy val ctCalibanServer: TaskKey[Unit] = taskKey[Unit]("Plugin configuration keys namespace")
+
+    // ## Required Plugin configurations
+    lazy val ctCalibanServerSettings: SettingKey[Seq[(String, ClientGenerationSettings)]] =
+      settingKey[Seq[(String, ClientGenerationSettings)]](
+        "(Required) List of GraphQL API reference and client generation configuration"
+      )
+
+    // ## Plugin task
+    lazy val ctCalibanServerGenerate: TaskKey[Seq[File]] = taskKey[Seq[File]]("Internal task")
+  }
+  import autoImport._
+
+  private val helpMsg: String =
+    """
+      |Missing configuration for the "CompileTimeCalibanServerPlugin" plugin.
+      |
+      |You need to configure the `Compile / ctCalibanServer / ctCalibanServerSettings` setting.
+      |
+      |Here's an example of configuration:
+      |```
+      |lazy val server =
+      |  project
+      |    .in(file("modules/server"))
+      |    .enablePlugins(CompileTimeCalibanServerPlugin)
+      |    .settings(
+      |      Compile / ctCalibanServer / ctCalibanServerSettings ++=
+      |        Seq(
+      |          "io.example.reference.GraphQLApi.api" -> ClientGenerationSettings.default,
+      |          "io.example.reference.GraphQLApi.api" ->
```

Comment:
Maybe keep only 1 out of the 3 for the example? It feels a bit confusing since in practice you would probably not have more than 1 for the same API.

Reply:
I'd prefer to keep them all here. This is not an example, this is a project used as a test. I need to keep something "complex" to ensure that the plugin is working correctly, otherwise, my tests would be too shallow

Reply:
Ah but I meant only the "help" message, not the actual code. But the most important is the actual documentation, this is less important to me.

Reply:
My bad. I'll change that. Thanks

Reply:
Done